c7n/resources/s3.py

```diff
@@ -447,6 +447,46 @@ def __call__(self, b):
             return False
         return True
 
+@filters.register('missing-lifecycle-rule')
+class MissingLifecycleRuleFilter(Filter):
+    """Find buckets missing a set of named Lifecycle rules.
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: s3-bucket-missing-lifecycle-rule
+                resource: s3
+                filters:
+                  - type: missing-lifecycle-rule
+                    rule_names:
+                      - AbortIncompleteMultipartUpload
+    """
+
+    schema = type_schema(
+        'missing-lifecycle-rule',
+        aliases=('missing-lifecycle',),
+        rule_names={'type': 'array', 'items': {'type': 'string'}})
```

Comment:
its not actually a rule name, the name is effectively the rule ID, this is a rule attribute presence detector. which makes think that this would be better done as a ValueFilter subclass that evaluated rules so people could match the actual attributes, ie 

```
 - type: lifecycle-rule
    key: AbortIncompleteMultipartUpload.DaysAfterInitialization
    value: 30
    op: less-than
```